docker/oap-es7/docker-entrypoint.sh

```diff
@@ -294,202 +33,51 @@ validateVariables() {
-   authentication: \${SW_AUTHENTICATION:""}
-   gRPCSslEnabled: \${SW_RECEIVER_SHARING_GRPC_SSL_ENABLED:false}
-   gRPCSslKeyPath: \${SW_RECEIVER_SHARING_GRPC_SSL_KEY_PATH:""}
-   gRPCSslCertChainPath: \${SW_RECEIVER_SHARING_GRPC_SSL_CERT_CHAIN_PATH:""}
-receiver-register:
-  default:
-receiver-trace:
-  default:
-    bufferPath: \${SW_RECEIVER_BUFFER_PATH:../trace-buffer/}  # Path to trace buffer files, suggest to use absolute path
-    bufferOffsetMaxFileSize: \${SW_RECEIVER_BUFFER_OFFSET_MAX_FILE_SIZE:100} # Unit is MB
-    bufferDataMaxFileSize: \${SW_RECEIVER_BUFFER_DATA_MAX_FILE_SIZE:500} # Unit is MB
-    bufferFileCleanWhenRestart: \${SW_RECEIVER_BUFFER_FILE_CLEAN_WHEN_RESTART:false}
-    sampleRate: \${SW_TRACE_SAMPLE_RATE:10000} # The sample rate precision is 1/10000. 10000 means 100% sample in default.
-    slowDBAccessThreshold: \${SW_SLOW_DB_THRESHOLD:default:200,mongodb:100} # The slow database access thresholds. Unit ms.
-receiver-jvm:
-  default:
-receiver-clr:
-  default:
-service-mesh:
-  default:
-    bufferPath: \${SW_SERVICE_MESH_BUFFER_PATH:../mesh-buffer/}  # Path to trace buffer files, suggest to use absolute path
-    bufferOffsetMaxFileSize: \${SW_SERVICE_MESH_OFFSET_MAX_FILE_SIZE:100} # Unit is MB
-    bufferDataMaxFileSize: \${SW_SERVICE_MESH_BUFFER_DATA_MAX_FILE_SIZE:500} # Unit is MB
-    bufferFileCleanWhenRestart: \${SW_SERVICE_MESH_BUFFER_FILE_CLEAN_WHEN_RESTART:false}
-istio-telemetry:
-  default:
-query:
-  graphql:
-    path: \${SW_QUERY_GRAPHQL_PATH:/graphql}
-alarm:
-  default:
-EOT
-    # generate telemetry
-    case ${SW_TELEMETRY} in
-    none) generateTelemetryNone;;
-    prometheus) generateTelemetryPrometheus;;
-    so11y) generateTelemetrySo11y;;
-    esac
+validateVariables "SW_CONFIGURATION" "$SW_CONFIGURATION" "none apollo nacos zookeeper"
 
-    # generate configuration
-    case ${SW_CONFIGURATION} in
-    none) generateConfigurationNone;;
-    apollo) generateConfigurationApollo;;
-    nacos) generateConfigurationNacos;;
-    zookeeper) generateConfigurationZookeeper;;
-    consul) generateConfigurationConsul;;
-    grpc) generateConfigurationGRPC;;
-    esac
+validateVariables "SW_TELEMETRY" "$SW_TELEMETRY" "none prometheus so11y"
 
-    cat <<EOT >> ${var_application_file}
-envoy-metric:
-  default:
-EOT
-    if [[ "$SW_ENVOY_ALS_ENABLED" = "true" ]]; then
-        cat <<EOT >> ${var_application_file}
-    alsHTTPAnalysis: \${SW_ENVOY_METRIC_ALS_HTTP_ANALYSIS:k8s-mesh}
-EOT
-    fi
-
-    if [[ "$SW_RECEIVER_ZIPKIN_ENABLED" = "true" ]]; then
-        cat <<EOT >> ${var_application_file}
-receiver_zipkin:
-  default:
-    host: \${SW_RECEIVER_ZIPKIN_HOST:0.0.0.0}
-    port: \${SW_RECEIVER_ZIPKIN_PORT:9411}
-    contextPath: \${SW_RECEIVER_ZIPKIN_CONTEXT_PATH:/}
-EOT
-    fi
+if [[ "$SW_ENVOY_ALS_ENABLED" = "true" ]]; then
+    export SW_ENVOY_METRIC_ALS_HTTP_ANALYSIS=k8s-mesh
+    echo "Set SW_ENVOY_METRIC_ALS_HTTP_ANALYSIS to ${SW_ENVOY_METRIC_ALS_HTTP_ANALYSIS}"
+fi
```

Comment:
I suggest removing these codes. I should provide some "backwards" compbility, just like "SW_STORAGE", but keeping it only add complex and confusing variables convertion.